Documentation for Preview Dataset (#1757)
kedro-org/kedro-viz: docs/source/preview_datasets.md

# Preview datasets in Kedro-Viz

> To provide users with a glimpse of their datasets within a Kedro project, Kedro-Viz offers a preview feature. This feature was introduced in Kedro-Viz version 6.3.0 and expanded upon in version 8.0.0. Initially, it supported `CSVDatasets` and `ExcelDatasets`, and later extended to encompass additional dataset types such as `PlotlyDatasets` and image datasets like `MatplotlibWriter`.

Reviewer: [Nit] `ExcelDatasets`, and **was** later extended to ..

Currently, Kedro-Viz supports four types of previews:

1. **TablePreview:** For datasets returning tables/dataframes.
2. **JSONPreview:** For datasets returning JSON objects.
3. **PlotlyPreview:** For datasets returning Plotly JSON objects.
4. **ImagePreview:** For datasets returning base64-encoded image strings.

> Whilst we currently support the above datasets. We are soon going to extend this functionality to other datasets. Users with custom datasets can also extend the preview functionality and we will cover that in the following sections. 

Reviewer: While we currently support the aforementioned **datasets, we** are soon going to extend this functionality to include other datasets. Users with custom datasets can also expand the preview functionality, and we will cover that in the following sections.

**Preview Tabular Data**

The page titled [Preview Tabular Data in Kedro-viz](./preview_pandas_datasets.md) contains a spaceflight tutorial that explains how you can enable preview on Tabular datasets such as `pandas.CSVDataset` and `pandas.ExcelDataset`.

**Preview Plotly Charts**

The page titled [Preview Plotly charts in Kedro-viz](./preview_plotly_datasets.md) contains a spaceflight tutorial that explains how you can create interactive visualizations using `PlotlyDatasets` on Kedro-viz.

**Preview Matplotlib Charts**

The page titled [Preview Matplotlib charts in Kedro-viz](./preview_matplotlib_datasets.md) contains a spaceflight tutorial that explains how you can create static visualizations using `MatplotlibWriterDataset` on Kedro-viz.

**Extend Preview to Custom Datasets**

> The page titled [Extend Preview to Custom Datasets](./preview_custom_datasets.md) contains information on how you can set up preview for custom datasets and what types are supported by Kedro-viz.

Reviewer: The page titled [Extend Preview to Custom Datasets](./preview_custom_datasets.md) contains information on **how to set up previews for custom datasets and which types are supported by Kedro-Viz**

```{toctree}
:maxdepth: 1
:hidden:
preview_matplotlib_datasets
preview_plotly_datasets
preview_pandas_datasets
preview_custom_datasets
```

## Disabling Previews

> To disable dataset previews for specific datasets, you need to set preview: false under the kedro-viz key within the metadata section of your conf.yml file. Here's an example configuration:

Reviewer: should we mention that previewing is made default from latest version of viz ? If someone is using old version of viz, it will be opt in ? Also, it will be a good idea to mention about the kedro-datasets version this new feature of kedro viz supports somewhere in the doc. Thank you

Author: good point.

# Preview datasets in Kedro-Viz

> To provide users with a glimpse of their datasets within a Kedro project, Kedro-Viz offers a preview feature. This feature was introduced in Kedro-Viz version 6.3.0 and expanded upon in version 8.0.0. Initially, it supported `CSVDataset` and `ExcelDataset`, and was later extended to encompass additional dataset types such as `PlotlyDataset` and image datasets like `MatplotlibWriter`.

Author: that makes sense.

> To provide users with a glimpse of their datasets within a Kedro project, Kedro-Viz offers a preview feature. This feature was introduced in Kedro-Viz version 6.3.0 and expanded upon in version 8.0.0. Initially, it supported `CSVDatasets` and `ExcelDatasets`, and was later extended to encompass additional dataset types such as `PlotlyDatasets` and image datasets like `MatplotlibWriter`.

Reviewer 2: ```suggestion
To provide users with a glimpse of their datasets within a Kedro project, Kedro-Viz offers a preview feature. This feature was introduced in Kedro-Viz version 6.3.0 and expanded upon in version 8.0.0. Initially, it supported `CSVDataset` and `ExcelDataset`, and was later extended to encompass additional dataset types such as `PlotlyDataset` and image datasets like `MatplotlibWriter`.
```

Currently, Kedro-Viz supports four types of previews:

1. **TablePreview:** For datasets returning tables/dataframes.
2. **JSONPreview:** For datasets returning JSON objects.
3. **PlotlyPreview:** For datasets returning Plotly JSON objects.
4. **ImagePreview:** For datasets returning base64-encoded image strings.

> While we currently support the aforementioned datasets, we are soon going to extend this functionality to include other datasets. Users with custom datasets can also expand the preview functionality, , which is covered in the section [Extend Preview to Custom Datasets](./preview_custom_datasets.md).

Reviewer 3: Do we have Vale enabled on this repo? I get the sense that _aforementioned_ would be flagged as too wordy

Author: i don't think so.

```{note}
Starting from Kedro-Viz version 8.0.0, the preview functionality on Kedro-Viz is now opt-out. For previous versions, this is an opt-in mechanism and users are required to specify `preview-args` for the preview to be enabled.

> By default, preview is now enabled for datasets. If you wish to disable preview for datasets, please refer to the [Disable Preview section](./preview_datasets.md#disabling-previews) for instructions.

Reviewer 3: What about

"Starting from Kedro-Viz 8.0.0 preview is now enabled for all datasets. If you wish to disable it for a specific dataset, refer to the [Disable Preview section](./preview_datasets.md#disabling-previews) for instructions."

```

**Preview Tabular Data**

> The page titled [Preview Tabular Data in Kedro-Viz](./preview_pandas_datasets.md) contains a spaceflight tutorial that explains how you can enable preview on Tabular datasets such as `pandas.CSVDataset` and `pandas.ExcelDataset`.

Reviewer 3: ```suggestion
See [Preview tabular data in Kedro-Viz](./preview_pandas_datasets.md) for an guide on how you can enable preview on tabular datasets such as `pandas.CSVDataset` and `pandas.ExcelDataset`.
```

**Preview Plotly Charts**

> The page titled [Preview Plotly charts in Kedro-Viz](./preview_plotly_datasets.md) contains a spaceflight tutorial that explains how you can create interactive visualizations using `PlotlyDataset` on Kedro-Viz.

Reviewer 3: ```suggestion
See [Preview Plotly charts in Kedro-Viz](./preview_plotly_datasets.md) for a guide on how you can create interactive visualizations using `PlotlyDataset` on Kedro-Viz.
```

**Preview Matplotlib Charts**

> The page titled [Preview Matplotlib charts in Kedro-Viz](./preview_matplotlib_datasets.md) contains a spaceflight tutorial that explains how you can create static visualizations using `MatplotlibWriterDataset` on Kedro-Viz.

Reviewer 3: (As above)

The page titled [Preview Matplotlib charts in Kedro-viz](./preview_matplotlib_datasets.md) contains a spaceflight tutorial that explains how you can create static visualizations using `MatplotlibWriterDataset` on Kedro-viz.

```yaml
companies:
  type: pandas.CSVDataSet
  filepath: data/01_raw/companies.csv
  metadata:
    kedro-viz:
      layer: raw
      preview_args:
          nrows: 5
**Extend Preview to Custom Datasets**

reviews:
  type: pandas.CSVDataSet
  filepath: data/01_raw/reviews.csv
  metadata:
    kedro-viz:
      layer: raw
      preview_args:
          nrows: 10
The page titled [Extend Preview to Custom Datasets](./preview_custom_datasets.md) contains information on how to set up previews for custom datasets and which types are supported by Kedro-Viz.

shuttles:
  type: pandas.ExcelDataSet
  filepath: data/01_raw/shuttles.xlsx
  metadata:
    kedro-viz:
      layer: raw
      preview_args:
          nrows: 15
```{toctree}
:maxdepth: 1
:hidden:
preview_matplotlib_datasets
preview_plotly_datasets
preview_pandas_datasets
preview_custom_datasets
```

> ## Previewing Data on Kedro-viz

Author: this entire section has now moved to `Preview Tabular Data on Kedro-viz`

> The page titled [Extend Preview to Custom Datasets](./preview_custom_datasets.md) contains information on how to set up previews for custom datasets and which types are supported by Kedro-Viz.

Reviewer 3: (As above)

> To disable dataset previews for specific datasets, you need to set `preview: false` under the `kedro-viz` key within the `metadata` section of your `conf.yml` file. Here's an example configuration:

Reviewer 3: ```suggestion
To disable dataset previews for specific datasets, you need to set `preview: false` under the `kedro-viz` key within the `metadata` section of your `catalog.yml` file. Here's an example configuration:
```
